packages/frontend/src/components/Game/GameBoard/index.jsx

```diff
@@ -1,17 +1,26 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import useGameContext from 'hooks/useGameContext';
 import ScoreBoard from '../ScoreBoard';
 import TileGrid from './TileGrid';
-
+import { handleGameOver } from 'contexts/game/actions';
+import ResultDisplay from './ResultDisplay';
 const GameBoard = () => {
   const {
-    state: { players, currentPlayer, tiles },
+    state: { players, currentPlayer, tiles, stage },
+    dispatch,
   } = useGameContext();
 
+  useEffect(() => {
+    const matchedTiles = tiles.filter((tile) => tile.isMatched === true);
+    if (tiles.length !== 0 && matchedTiles.length === tiles.length) {
+      dispatch(handleGameOver());
+    }
+  }, [tiles]);
+
```

Comment:
What do you think about adding the `matchedTiles` array to the context instead of calculating it here?

Reply:
I thought that having matchedTiles calculated in the useEffect in this component was a cleaner way than calculating it in the HANDLE_FLIPPED_PAIR section of the reducer logic. Calculated in the reducer logic and then reading it from the context like this:
![Screen Shot 2024-03-01 at 1 44 29 PM](https://github.com/perennialAutodidact/memory-snap/assets/81979567/b563d4d7-15e3-44d2-a041-274c49f60460)
![Screen Shot 2024-03-01 at 1 46 57 PM](https://github.com/perennialAutodidact/memory-snap/assets/81979567/7ab7fcb3-7b15-4e26-9e68-e90095a2be15)

the matchedTiles state is always a step behind the game.

Reply:
Apologies, are you saying that calculating the matched tiles in the reducer makes the array one step behind and calculating it in the `useEffect`, which makes it update at the correct time?

Reply:
Yes, but thats just with the code written like I've shown there. 

I'm wondering if its a similar situation to what we ran into in the flip tile logic? There we needed to make a temp state variable to have a flipped array that reflected the length of state as well as the incoming payload to properly set the lockedTiles helper all within the same action. 

Just trying to figure this out. It's probably simpler than I'm seeing.

Reply:
I'd have to dive into the code a bit more and play with some values to figure out exactly what's going on. If you push your changes I can check it out in a bit

Reply:
@perennialAutodidact changes pushed.

Reply:
@perennialAutodidact This works, check it out:
![Screen Shot 2024-03-01 at 7 09 49 PM](https://github.com/perennialAutodidact/memory-snap/assets/81979567/603ef9fb-ab1c-435a-b16c-d2900480f25e)


Just needed to make that change to the tiles array before the return.

Reply:
Nice! That's exactly the type of arrangement I was thinking of as well. I feel like there could be a helper function for lines 54-56, since they're identical to lines 61-63. Could make things a bit more readable.

Reply:
I'm thinking, another way to not repeat code would be to just use the value of tempState.tiles on line 59. I was repeating the code when I really only need to repeat the value. 

![Screen Shot 2024-03-04 at 11 04 24 AM](https://github.com/perennialAutodidact/memory-snap/assets/81979567/f03b3461-ac70-47ab-acad-5911facb39b3)


To make a helper function, I would import isMatchingPair, handleMatch, and resetTiles into a helper function called handleFlippedTiles and use them there. Since we don't have to repeat code in the above option though, I think it's more readable to have those functions still appear in the reducer logic because of their descriptive names. You can see whats going on instead of going and looking at what is going on in the helper function. 

Let me know what you think.

Reply:
That's fine with me. We may want to reassess the situation if there ends up being a third place where that logic needs to happen.